pytorch_lightning/core/memory.py

```diff
@@ -182,14 +189,25 @@ class ModelSummary(object):
         0.530     Total estimated model params size (MB)
     """
 
-    MODE_TOP = "top"
-    MODE_FULL = "full"
-    MODE_DEFAULT = MODE_TOP
-    MODES = [MODE_FULL, MODE_TOP]
+    MODES = dict(top=1, full=-1)  # TODO: remove in v1.6
 
-    def __init__(self, model, mode: str = MODE_DEFAULT):
+    def __init__(self, model, mode: Optional[str] = "top", max_depth: Optional[int] = None):
         self._model = model
-        self._mode = mode
+
+        #  temporary mapping from mode to max_depth
+        if max_depth is None:
+            if mode in ModelSummary.MODES:
+                from pytorch_lightning.utilities import rank_zero_deprecation
+                rank_zero_deprecation(
+                    "Argument `mode` in `ModelSummary` is deprecated in v1.4"
+                    " and will be removed in v1.5. Use `max_depth` instead."
+                )
+                max_depth = ModelSummary.MODES[mode]
+            else:
+                from pytorch_lightning.utilities.exceptions import MisconfigurationException
+                raise MisconfigurationException(f"`mode` can be {', '.join(ModelSummary.MODES)}, got {mode}.")
```

Comment:
To mimic the previous behaviour, this should set `max_depth = 1` when the user does not pass any argument